src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -3178,6 +3178,58 @@ public GHTagObject createTag(String tag, String message, String object, String t
                 .wrap(this);
     }
 
+    /**
+     * Lists the code scanning alerts of this repository.
+     *
+     * @return the paged iterable
+     */
+    public PagedIterable<GHCodeScanningAlert> listCodeScanningAlerts() {
+        return listCodeScanningAlerts(Collections.emptyMap());
+    }
+
+    /**
+     * Lists the code scanning alerts of this repository filtered on the alert status
+     *
+     * @param state
+     *            alert status to filter on
+     * @return the paged iterable
+     */
+    public PagedIterable<GHCodeScanningAlert> listCodeScanningAlerts(GHCodeScanningAlertState state) {
+        return listCodeScanningAlerts(Collections.singletonMap("state", state.name().toLowerCase()));
+    }
+
+    /**
+     * Lists the code scanning alerts of this repository filtered on the code scanning tool name
+     *
+     * @param toolName
+     *            name of code scanning tool that creates alerts
+     * @return the paged iterable
+     */
+    public PagedIterable<GHCodeScanningAlert> listCodeScanningAlerts(String toolName) {
+        return listCodeScanningAlerts(Collections.singletonMap("tool_name", toolName));
+    }
```

Comment:
There are other filters: 
https://docs.github.com/en/rest/reference/code-scanning#list-code-scanning-alerts-for-a-repository

These filters can be combined so having only single filter method available doesn't seem right.